pkg/distributor/distributor.go

```diff
@@ -207,19 +259,23 @@ func (d *Distributor) Push(ctx context.Context, req *connect.Request[pushv1.Push
 		for _, raw := range series.Samples {
 			usagestats.NewCounter(fmt.Sprintf("distributor_profile_type_%s_received", profName)).Inc(1)
 			d.profileReceivedStats.Inc(1)
-			d.bytesReceivedTotalStats.Inc(int64(len(raw.RawProfile)))
-			d.bytesReceivedStats.Record(float64(len(raw.RawProfile)))
+			if haveRawPprof {
+				d.metrics.receivedCompressedBytes.WithLabelValues(profName, tenantID).Observe(float64(len(raw.RawProfile)))
+			}
 			totalProfiles++
-			d.metrics.receivedCompressedBytes.WithLabelValues(profName, tenantID).Observe(float64(len(raw.RawProfile)))
-			p, err := pprof.RawFromBytes(raw.RawProfile)
-			if err != nil {
-				return nil, connect.NewError(connect.CodeInvalidArgument, err)
+			p := raw.Profile
+			var decompressedSize int
+			if haveRawPprof {
+				decompressedSize = p.SizeBytes()
+			} else {
+				decompressedSize = p.SizeVT()
```

Comment:
What's the difference between `p.SizeBytes()` and `p.SizeVT()`. Sounds like SizeVT is more accurate ? If so may be we should always use it.

Reply:
the difference is `SizeBytes() ` is just `len(buf)` and `SizeVT` does the deep structural size computation
for jfr `len(buf) == 0`

Reply:
so we need to calculate it

Reply:
yeah I was wondering how much overhead SizeVT() and if we should consider just calling it all the time. I think it's fine to keep how you did.

Reply:
resolving this as I reverted distributor changes